packages/core-modules/contracts/storage/ElectionStorage.sol

```diff
@@ -15,45 +15,46 @@ contract ElectionStorage {
 
     struct ElectionData {
         /**
-         * @dev History of all the votes casted by voter addrees
+         * @dev Compilation of all the ballots casted by voters. Works together with ballotsIndexes
          */
-        VoteData[] votes;
+        Ballot[] ballots;
         /**
-         * @dev Position of an address on the votes Array.
+         * @dev Position of an address (voter) on the ballots Array.
          */
-        mapping(address => uint256) votesIndexes;
+        mapping(address => uint256) ballotsIndex;
         /**
-         * @dev number of votes a nominee has for being a council member in the next epoch.
+         * @dev number of votes a candidate has for being a council member in the next epoch.
+         * @dev Note the interpretation is dependant on the voting strategy
          */
-        mapping(address => uint256) nomineeVotes;
+        mapping(address => uint256) candidateVotes;
         /**
          * @dev History of addresses that voted on any of the elections.
          */
         mapping(address => bool) addressVoted;
         /**
-         * @dev Used to keep track of the next epoch's nominees. Gets erased when an epoch starts and a new council takes effect.
+         * @dev Used to keep track of the next epoch's nominees.
          */
-        address[] nominees;
+        address[] candidates;
+        /**
+         * @dev Position of an address on the candidates Array. Note that position 1 corresponds to index 0.
+         */
+        mapping(address => uint256) candidatePositions;
         /**
-         * @dev Position of an address on the nominees Array. Note that position 1 corresponds to index 0.
+         * @dev Used to keep track of the latest candidate processed in the last batch.
          */
-        mapping(address => uint256) nomineePositions;
+        uint256 processedBatchIndex;
         /**
          * @dev Flag to indicate if the election was evaluated (if batched, latest batch was processed).
          */
         bool isElectionEvaluated;
         /**
-         * @dev Used to keep track of the next epoch's nextEpochMembers. Gets erased when an epoch starts and a new council takes effect.
+         * @dev Used to keep track of the next epoch's nextEpochMembers.
```

Comment:
Updated comments. It won't get erased but a new, clean, ElectionData is used per election